### PR TITLE
[TASK] Modernize Preview middleware and omit usage of GLOBALS

### PR DESCRIPTION
**Pull request by ervaude**
Closes: #19
Closes: #7

### CODE REVIEW

File: Classes/Http/Middleware/Preview.php [hunk 1]
```diff
@@ -73,24 +73,23 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
             $normalizedParams = $request->getAttribute('normalizedParams');
             $cookie = new Cookie(
-                PreviewUriBuilder::PARAMETER_NAME,
-                $hash,
-                0,
-                $normalizedParams->getSitePath(),
-                '',
-                true,
-                true
+                name: PreviewUriBuilder::PARAMETER_NAME,
```
Comment (bmack) [inline Classes/Http/Middleware/Preview.php:88]: sorry, this is PHP 8.x feature, but the current code supports also PHP 7.4

File: Classes/Http/Middleware/Preview.php [hunk 2]
```diff
@@ -34,16 +35,15 @@ class Preview implements MiddlewareInterface
 {
     public const REQUEST_ATTRIBUTE = 'tx_authorized_preview_config';
 
-    protected Context $context;
-
-    public function __construct(Context $context)
-    {
-        $this->context = $context;
-    }
+    public function __construct(protected readonly Context $context) {}
```
Comment (bmack) [inline Classes/Http/Middleware/Preview.php:38]: sorry, this is PHP 8.x feature, but the current code supports also PHP 7.4

**Review by bmack — CHANGES_REQUESTED** [review]

File: Classes/Http/Middleware/Preview.php (hunk 1 above)
Comment (ervaude) [reply Classes/Http/Middleware/Preview.php:88]: not anymore we dont 😀

File: Classes/Http/Middleware/Preview.php (hunk 2 above)
Comment (ervaude) [reply Classes/Http/Middleware/Preview.php:38]: not anymore we dont 😀

**Review by bmack — APPROVED** [review]

**Pull request merged**